### DISCUSSION TITLE
Hubs-Cloud: SSL certificate error when running local client

### DISCUSSION THREAD

**Discussion by leonardoazzi** [mozilla/hubs#5173] [Hubs Cloud Questions] [answered]
**Description**
Since we had an AWS template update, I started to get some SSL certificate errors when I try to run a local client against Hubs Cloud. This issue raises a general error when loading any scene.

I merged some new code and solved some other minor issues, as:
- hubs-cloud branch
- murphy's PR (https://github.com/mozilla/hubs/pull/5125)
- @markusTraber workaround (https://github.com/mozilla/hubs/issues/5150)

**To Reproduce**
1. Checkout to hubs-cloud branch;
2. Run `npm run login` and authenticate with your hubs cloud;
3. `git pull` to get the newest code;
4. Create a new branch from hubs-cloud
5. Merge murphy's and markus' PRs (isolate just the SSL error);
6. Run `npm start`
7. Create or load any room.

**Expected behavior**
The browser should raise a SSL certificate error.

**Screenshots**
![image](https://user-images.githubusercontent.com/13557537/154503818-96c530fc-50b5-45c1-a55d-5a24391ee566.png)
![image](https://user-images.githubusercontent.com/13557537/154503832-94cdf55d-6881-44ca-96af-d2fb61d12be9.png)

**Hardware** 
 - Device: Desktop
 - OS: Pop!_OS 21.10
 - Browser: Chromium 98.0.4758.80 

**Additional context**
Cloudflare is not active.
P.S.: I've got some CORS errors when trying with `npm run dev`

┆Issue is synchronized with this [Jira Task](https://mozilla-hub.atlassian.net/browse/HUBS-1279)

**Comment by leonardoazzi** [comment, marked as answer]
We've got it fixed in #5136 . :)
Also @robinkwilson sent me some instructions to accept the self-signed cert from my IP's as the following tutorial: https://youtu.be/l33b5GTrC5Q
So, closing this issue. Thanks you Hubs team and community <3

**Comment by NigelRidderhof** [comment]
Hi! 

We've pulled the live Hubs Cloud code yesterday and started merging this with our custom client code. So this was after the code that fixed this problem for you was merged into the Hubs Cloud branch. But I am still having similar problems that you've described above when using `npm run dev`. 

Before this attempt to update the code, `npm run dev` worked just fine on https://localhost:8080/, but now the scene doesn't load and I get these SSL certificate errors as well. 
![image](https://user-images.githubusercontent.com/43579707/155160520-d365c084-5199-424f-8633-39b2728d48dd.png)

I performed a `npm ci`, a `npm run login` to run `npm run dev` again, but that didn't help.

I then started reading through discuissions like this https://github.com/mozilla/hubs/discussions/4888 
Before this update I used a custom CORS proxy server link in my `.defaults.env`. Following the tips in this discussion I changed this to `CORS_PROXY_SERVER="hubs-proxy.com"` and started running on `https://hubs.local:8080/`. In this case the scene sort of loads, but the frame is immediately frozen - and the same erros occur.
![image](https://user-images.githubusercontent.com/43579707/155152054-b85ae4f5-6fe0-437e-882c-ae533966e82c.png)

I also gtried adding `HOST_IP=hubs.local` to `.defaults.env` which I read in the same discussion, but this didn't help and might be just a `npm start` thing. 
Sidenote: actually running `npm start` (which is new to me) doesn't gets me as close as `npm run dev` does, and gives 401 api errors and more below. 

Hope someone know's what to do here :).
@[leonardoazzi](https://github.com/leonardoazzi)

**Reply by leonardoazzi** [reply]
> Hi!
> 
> We've pulled the live Hubs Cloud code yesterday and started merging this with our custom client code. So this was after the code that fixed this problem for you was merged into the Hubs Cloud branch. But I am still having similar problems that you've described above when using `npm run dev`.
> 
> Before this attempt to update the code, `npm run dev` worked just fine on https://localhost:8080/, but now the scene doesn't load and I get these SSL certificate errors as well. ![image](https://user-images.githubusercontent.com/43579707/155160520-d365c084-5199-424f-8633-39b2728d48dd.png)
> 
> I performed a `npm ci`, a `npm run login` to run `npm run dev` again, but that didn't help.
> 
> I then started reading through discuissions like this #4888 Before this update I used a custom CORS proxy server link in my `.defaults.env`. Following the tips in this discussion I changed this to `CORS_PROXY_SERVER="hubs-proxy.com"` and started running on `https://hubs.local:8080/`. In this case the scene sort of loads, but the frame is immediately frozen - and the same erros occur. ![image](https://user-images.githubusercontent.com/43579707/155152054-b85ae4f5-6fe0-437e-882c-ae533966e82c.png)
> 
> I also gtried adding `HOST_IP=hubs.local` to `.defaults.env` which I read in the same discussion, but this didn't help and might be just a `npm start` thing. Sidenote: actually running `npm start` (which is new to me) doesn't gets me as close as `npm run dev` does, and gives 401 api errors and more below.
> 
> Hope someone know's what to do here :). @[leonardoazzi](https://github.com/leonardoazzi)

Hi, Nigel. 

Have you tried to follow this tutorial? https://youtu.be/l33b5GTrC5Q
It solved my issue with the scene not loading, after I merged the new hubs cloud code.

Hope it helps!

**Reply by NigelRidderhof** [reply]
Thanks for getting back to me! I  actually did watch that video ealier when I saw you mentioning it, but I don't get that exact error. And I think you used `npm start` instead of `npm run dev` right? I didn't see that error inside you console errors either, so I didn't think I could do anything with it that tutorial. Let me know if I'm misunderstanding something!

**Reply by leonardoazzi** [reply]
Oh, too bad. :(
The command `npm run dev` relies on a Mozilla's dev server. But maybe it gone old and lost in the documentation. I know that the easiest way to run a custom client is against a Hubs Cloud instance, which command is `npm start`. I've done some quick test and I've got the same issue with `run dev` here, while against my Hubs Cloud works fine.

**Reply by NigelRidderhof** [reply]
Aah thank you for these insights! I now switched over to `npm run start`, fixed some errors but now I run into the same errors in here. The scene seems to load but is immediately stuck. How did you get from having these errors you see below and in you screenshots to having that "Cross-Origin Request Blocked" that you can fix with the YouTube video? Did you merge a specific commit?

![image](https://user-images.githubusercontent.com/43579707/155296768-3f481848-ed80-470a-b6ea-30624a489237.png)

**Reply by leonardoazzi** [reply]
> Aah thank you for these insights! I now switched over to `npm run start`, fixed some errors but now I run into the same errors in here. The scene seems to load but is immediately stuck. How did you get from having these errors you see below and in you screenshots to having that "Cross-Origin Request Blocked" that you can fix with the YouTube video? Did you merge a specific commit?
> 
> ![image](https://user-images.githubusercontent.com/43579707/155296768-3f481848-ed80-470a-b6ea-30624a489237.png)

Nice!!
I think I still get this errors in my console, but everything works fine. It's not to worry about, at least until now 😅. Are you getting another error? Can you send a screenshot with the complete log?

**Reply by NigelRidderhof** [reply]
Ahh really, that was very good to know 'cause now I started looking at the logs underneath. Much lower there was a very hidden problem which was no red error but did break it, coming from our custom code. My `npm run start` is now running and now I can start fixing things up after the merge. I thank you with all of my heart!!

**Reply by leonardoazzi** [reply]
Yay! You're welcome! 🦆
